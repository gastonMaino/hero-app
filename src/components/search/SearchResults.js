import React, { useMemo } from 'react';
import { HeroeContext } from '../heroes/HeroContext';
import { HeroCard } from '../heroes/HeroCard';
import { getHeroesByName } from '../../selectors/getHeroesByName';
import PropTypes from 'prop-types';

export const SearchResults = ({q}) => {

    const heroData = useMemo(() => getHeroesByName(q), [q]);


    return (
        <section className='section-results'>
            <h2>Results: </h2>

                {
                    (q === '')
                    &&
                    <div className='alert'>
                        <p className='alert__paragraph'>
                            Search a hero!
                        </p>
                    </div>
                }

                {
                    (q !== '' && heroData.length === 0)
                    &&
                    <div className='alert-danger'>
                        <p className='alert-danger__paragraph'>
                            There is no a hero {q}
                        </p>
                    </div>
                }

                {
                    heroData.map((hero) => {
                        return (
                            <HeroeContext.Provider
                                key={hero.id} value={hero}
                            >
                                <HeroCard />
                            </HeroeContext.Provider>
                        )

                    })
                }    
        </section>
    )
}

SearchResults.propTypes = {
    q: PropTypes.string.isRequired
}
