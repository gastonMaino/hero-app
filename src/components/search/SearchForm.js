import React from 'react';
import queryString from 'query-string'
import { useLocation } from 'react-router';
import { useForm } from '../../hooks/useForm/useForm'
import { SearchResults } from './SearchResults';


export const SearchForm = ({ history }) => {


    const location = useLocation();
    const { q = '' } = queryString.parse(location.search);

    const [{ search }, handleInputChange] = useForm({
        search: q
    })

    const handleSearch = (e) => {
        e.preventDefault();
        history.push(`?q=${search}`)
    }

    return (
        <>
            <section className='search-form'>
                <h2 className='search-form__title'>Search Form:</h2>
                <form onSubmit={handleSearch} className='form'>
                    <input
                        type='text'
                        name='search'
                        value={search}
                        placeholder='find your hero'
                        aria-label='find your hero'
                        className='form__input'
                        autoComplete='off'
                        onChange={handleInputChange}
                    />
                    <button className='btn-search'>
                        Search
                    </button>
                </form>
            </section>

            <SearchResults q={q} />
        </>
    )
}
