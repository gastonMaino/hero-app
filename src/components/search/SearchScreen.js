import React from 'react';
import { SearchForm } from './SearchForm';

export const SearchScreen = ({ history }) => {
    return (
        <article className='search'>
            <div className='container-search-title'>
                <h1 className='search-title'>Search</h1>
            </div>

            <div className='container-items-search'>
                <SearchForm history={history} />
            </div>
        </article>
    )
}
