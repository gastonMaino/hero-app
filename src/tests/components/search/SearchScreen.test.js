import { shallow } from "enzyme"
import { SearchScreen } from "../../../components/search/SearchScreen"

describe('test in <SearchScreen />', () => {
    
    test('should match to snapshot', () => {
        const wrapper = shallow(<SearchScreen />);

        expect(wrapper).toMatchSnapshot();
    })
})
