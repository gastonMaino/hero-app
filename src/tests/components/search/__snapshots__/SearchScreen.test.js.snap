// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <SearchScreen /> should match to snapshot 1`] = `
<article
  className="search"
>
  <h1>
    Search
  </h1>
  <SearchForm />
</article>
`;
